### PR TITLE
Migrate HDInsight Management SDK to use the new data-plane URL recommended by Azure Data Lake

### PR DESCRIPTION
**Pull request by srsiva**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/dev/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@srsiva__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sisankar). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Review by shefaliv — APPROVED** [review]

File: src/ResourceManagement/HDInsight/HDInsight/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -32,8 +32,8 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 
-[assembly: AssemblyVersion("2.0.0.0")]
-[assembly: AssemblyFileVersion("2.0.0.0")]
+[assembly: AssemblyVersion("2.1.0.0")]
```
Comment (markcowl) [inline src/ResourceManagement/HDInsight/HDInsight/Properties/AssemblyInfo.cs]: AssemblyVersion should not change, for backward compatibility

**Review by markcowl — CHANGES_REQUESTED** [review]

File: src/ResourceManagement/HDInsight/HDInsight/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (srsiva) [inline src/ResourceManagement/HDInsight/HDInsight/Properties/AssemblyInfo.cs]: fixed.

File: src/ResourceManagement/HDInsight/HDInsight/CustomizationModels/ServicePrincipal.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ public ServicePrincipal(Guid applicationId, Guid aadTenantId, byte[] certificate
             this.CertificatePassword = certificatePassword;
 
             //Resource Uri of data lake 
-            this.ResourceUri = new Uri("https://management.core.windows.net/");
+            this.ResourceUri = new Uri("https://datalake.azure.net/");
```
Comment (markcowl) [inline src/ResourceManagement/HDInsight/HDInsight/CustomizationModels/ServicePrincipal.cs]: Please provide at least one updated test recording using the new resource uri

File: src/ResourceManagement/HDInsight/HDInsight/Microsoft.Azure.Management.HDInsight.nuget.proj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup>
     <SdkNuGetPackage Include="Microsoft.Azure.Management.HDInsight">
-      <PackageVersion>2.0.0</PackageVersion>
+      <PackageVersion>2.1.0</PackageVersion>
```
Comment (markcowl) [inline src/ResourceManagement/HDInsight/HDInsight/Microsoft.Azure.Management.HDInsight.nuget.proj]: Package version should be 2.0.1

**Review by markcowl — CHANGES_REQUESTED** [review]

File: src/ResourceManagement/HDInsight/HDInsight/Microsoft.Azure.Management.HDInsight.nuget.proj (hunk 3 above)
Comment (srsiva) [inline src/ResourceManagement/HDInsight/HDInsight/Microsoft.Azure.Management.HDInsight.nuget.proj]: fixed

File: src/ResourceManagement/HDInsight/HDInsight/CustomizationModels/ServicePrincipal.cs (hunk 2 above)
Comment (srsiva) [inline src/ResourceManagement/HDInsight/HDInsight/CustomizationModels/ServicePrincipal.cs]: fixed

**Review by markcowl — APPROVED** [review]

**Pull request merged**